File: src/main/java/binarysearch/LongestIntervalContainingOneNumber.java
package binarysearch;

import java.util.Arrays;

public class LongestIntervalContainingOneNumber {
    public int solve(int[] nums) {
        Arrays.sort(nums);
        int ms = Integer.MIN_VALUE;
        for (int i = 0; i < nums.length; i++) {
            int hi = i < nums.length - 1 ? nums[i + 1] - 1 : 100_000;
            int lo = i > 0 ? nums[i - 1] + 1 : 1;
            if (lo <= nums[i] && nums[i] <= hi) ms = Math.max(ms, Math.abs(hi - lo) + 1);
        }
        return ms;
    }
}

File: src/main/cpp/binarysearch/longest_interval_containing_one_number.cpp
#include <bits/stdc++.h>

using namespace std;

int solve(vector<int> &nums) {
	if (nums.empty()) return 0;
	nums.push_back(0);
	sort(nums.begin(), nums.end());
	nums.push_back(100'001);
	int ans = 0;
	for (int i = 2; i < nums.size(); i++) {
		ans = max(ans, nums[i] - nums[i - 2] - 1);
	}
	return ans;
}